### PR TITLE
Auto fill of curves table by selecting group curves names on Chart setup dialog box

### PR DESCRIPTION
**Pull request by demvlad**
**Improvement the Chart setup dialog box.**
_Current version behavior:_ by selecting group curves names (like "Gyros", "Motors", "PID P" and other) on Charts setup dialog box it is need to close dialog box to Apply changes and open it again to edit curves. 
_This PR behavior:_ the group curves are auto extended by select it in list. It is not need to close-open dialog box to edit these curves.

### CODE REVIEW

File: src/graph_config.js [hunk 1]
```diff
@@ -560,7 +560,7 @@ GraphConfig.load = function(config) {
                                         max: 50
                                     }
                                 };
-                            case 'debug[3]': // Vario
+                            case 'debug[3]': // letio
```
Comment (haslinghuis) [inline src/graph_config.js:563]: :) - this one should stay

File: src/graph_config.js [hunk 2]
```diff
@@ -44,18 +44,73 @@ export function GraphConfig(graphConfig) {
...
+    }
+
+    let adaptField = function(flightLog, field, colorIndexOffset, forceNewCurve) {
+        const defaultCurve = GraphConfig.getDefaultCurveForField(flightLog, field.name);
+        let colorIndex = 0;
+        if (field.curve === undefined || forceNewCurve) {
+            field.curve = defaultCurve;
+        } else {
+            if (field.curve.MinMax == undefined)
+                            field.curve.MinMax = defaultCurve.MinMax;
```
Comment (haslinghuis) [inline src/graph_config.js:81]: :space_invader: 

Please fix indentation.

Comment (demvlad) [reply src/graph_config.js]: Ok. Resolved.

**Comment by haslinghuis** [comment]
Please check sonar cloud issues.

**Comment by demvlad** [comment]
> Please check sonar cloud issues.

Ok. The main issues were resolved.

**Review by haslinghuis — DISMISSED** [review]

File: src/graph_config.js (hunk 1 above)
Comment (haslinghuis) [reply src/graph_config.js:554]: I mean the comment needs to be vario not letio

File: src/graph_config_dialog.js [hunk 3]
```diff
@@ -273,16 +285,18 @@ export function GraphConfigurationDialog(dialog, onSave) {
         $('select.graph-height', graphElem).replaceWith(chooseHeight(graph.height?(graph.height):1));
 
         // Add Field List
-        for (var i = 0; i < graph.fields.length; i++) {
-            var
-                field = graph.fields[i],
-                fieldElem = renderField(flightLog, field, field.color?(field.color):(GraphConfig.PALETTE[i].color));
-
-            fieldList.append(fieldElem);
+        for (let i=0; i<graph.fields.length; ++i) {
```
Comment (haslinghuis) [inline src/graph_config_dialog.js:288]: - Did you intent to start with ++i ?
- Operators needs space for better readability

```suggestion
        for (let i = 0;  i < graph.fields.length; i++) {
```

File: src/graph_config_dialog.js [hunk 4]
```diff
@@ -302,20 +316,17 @@ export function GraphConfigurationDialog(dialog, onSave) {
...
     function renderGraphs(flightLog, graphs) {
-        var
+        let
             graphList = $(".config-graphs-list", dialog);
 
         graphList.empty();
 
-        for (var i = 0; i < graphs.length; i++) {
+        for (let i=0; i<graphs.length; ++i) {
             graphList.append(renderGraph(flightLog, i, graphs[i]));
```
Comment (haslinghuis) [inline src/graph_config_dialog.js:325]: Same as above

File: src/graph_config_dialog.js [hunk 5]
```diff
@@ -326,25 +337,23 @@ export function GraphConfigurationDialog(dialog, onSave) {
             dividerAfter: true
         });
 
-        for (i = 0; i < exampleGraphs.length; i++) {
-            var
-                graph = exampleGraphs[i],
-                li = $('<li><a href="#"></a></li>');
+        for (let i=0; i < exampleGraphs.length; ++i) {
```
Comment (haslinghuis) [inline src/graph_config_dialog.js:340]: Same as above

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: src/graph_config_dialog.js (hunk 5 above)
Comment (demvlad) [reply src/graph_config_dialog.js:340]: I do not know about JS, but in C  "++i" is more optimal code then "i++"
The spaces - Ok.

Comment (haslinghuis) [reply src/graph_config_dialog.js:340]: i++ is the postfix increment operator, which increments the value of i after it is used in an expression. On the other hand, ++i is the prefix increment operator, which increments the value of i before it is used in an expression.

**Comment by haslinghuis** [comment]
@nerdCopter how is your regexp for remaining issues?

**Review by haslinghuis — APPROVED** [review]

File: src/graph_config_dialog.js (hunk 5 above)
Comment (demvlad) [reply src/graph_config_dialog.js]: > i++ is the postfix increment operator, which increments the value of i after it is used in an expression. On the other hand, ++i is the prefix increment operator, which increments the value of i before it is used in an expression.

i++ create tmp variable to store values, the ++i - is not.
In these cases it is not need to use unincrement value.
But for one code style - i've made change.

**Comment by demvlad** [comment]
> @nerdCopter how is your regexp for remaining issues?

about regexp, i know what is, but as honestly, i do not know the reg syntax :)

**Comment by nerdCopter** [comment]
> @nerdCopter how is your regexp for remaining issues?

i don't understand the problem to answer. :eyes:

**Comment by demvlad** [comment]
Improve curves color assignment.
Code style improvement.

**Pull request merged**